### ISSUE TITLE
e2e test failure: error: failed to create clusterrolebinding: clusterrolebindings.rbac.authorization.k8s.io "e2e-test-cluster-admin-binding" already exists

### ISSUE THREAD

**Issue by brianpursley** [kubernetes/kubernetes#94809]
##### Error text:
```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/autoscaling/custom_metrics_stackdriver_autoscaling.go:54
Sep 15 04:05:10.238: Failed to set up: error running /workspace/kubernetes/platforms/linux/amd64/kubectl --server=https://35.227.151.106 --kubeconfig=/workspace/.kube/config --namespace=horizontal-pod-autoscaling-8880 create clusterrolebinding e2e-test-cluster-admin-binding --clusterrole=cluster-admin --user=pr-kubekins@kubernetes-jenkins-pull.iam.gserviceaccount.com:
Command stdout:

stderr:
error: failed to create clusterrolebinding: clusterrolebindings.rbac.authorization.k8s.io "e2e-test-cluster-admin-binding" already exists

error:
exit status 1
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/autoscaling/custom_metrics_stackdriver_autoscaling.go:264
```
#### Recent failures:
[9/15/2020, 3:31:33 AM ci-kubernetes-e2e-gci-gce-autoscaling-hpa-cm](https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-autoscaling-hpa-cm/1305771124411338752)
[9/15/2020, 3:20:00 AM ci-kubernetes-e2e-gci-gce-autoscaling-migs-hpa](https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-autoscaling-migs-hpa/1305768225241305090)
[9/15/2020, 2:59:54 AM ci-kubernetes-e2e-gci-gce-autoscaling-hpa-cm](https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-autoscaling-hpa-cm/1305763192298803200)
[9/15/2020, 2:49:00 AM ci-kubernetes-e2e-gci-gce-autoscaling-migs-hpa](https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-autoscaling-migs-hpa/1305760424309821440)
[9/15/2020, 2:28:55 AM ci-kubernetes-e2e-gci-gce-autoscaling-hpa-cm](https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-autoscaling-hpa-cm/1305755390901751810)

/kind failing-test

/sig autoscaling

Looks like it just started happening today (@ 2020-09-15 03:54:19 +0000 UTC)
![image](https://user-images.githubusercontent.com/4977542/93255338-c8c0eb80-f767-11ea-991a-6f682cf69169.png)

**Comment by neolit123** [comment]
cc @soltysh @zhouya0
https://github.com/kubernetes/kubernetes/pull/94609

> The way our current kubectl e2e are written requires explicitly passing --namespace flag, even though namespace is one of the parameters passed to RunKubectlOrDie.

https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-autoscaling-hpa-cm/1305716383220240387

> the namespace from the provided object "custom-metrics" does not match the namespace "horizontal-pod-autoscaling-830". You must pass '--namespace=custom-metrics' to perform this operation.
the namespace from the provided object "kube-system" does not match the namespace "horizontal-pod-autoscaling-830". You must pass '--namespace=kube-system' to perform this operation.
the namespace from the provided object "custom-metrics" does not match the namespace "horizontal-pod-autoscaling-830". You must pass '--namespace=custom-metrics' to perform this operation.
the namespace from the provided object "custom-metrics" does not match the namespace "horizontal-pod-autoscaling-830". You must pass '--namespace=custom-metrics' to perform this operation.

**Comment by neolit123** [comment]
related fix:
https://github.com/kubernetes/kubernetes/pull/94789

**Comment by zhouya0** [comment]
@neolit123 No I don't think that's the fix for this (that is just a little cleanup). 
Actually I think we have to delete all namespace parameters in `CreateAdapter` and `createClusterAdminBinding` :
https://github.com/kubernetes/kubernetes/blob/965137a992ed755474f17fa6377ee3555745addd/test/e2e/instrumentation/monitoring/custom_metrics_deployments.go#L255-L271

Because before `framework.RunKubectl(namespace, "create", "-f", adapterURL)`, the `namespace` will have no actually effect. But now with the change of https://github.com/kubernetes/kubernetes/pull/94609, it will have namespace `horizontal-pod-autoscaling-830`:
```
	if tk.Namespace != "" {
		defaultArgs = append(defaultArgs, fmt.Sprintf("--namespace=%s", tk.Namespace))
	}
```
at the end of the command. But the namespace in yaml file is  `custom-metrics` which is different from `horizontal-pod-autoscaling-830`.
Basically when applying some resources from file, we should not have `--namespace` parameter, right?
Related PR: https://github.com/kubernetes/kubernetes/pull/84613
